### PR TITLE
Clarify logic in callback func in is authorized callback

### PR DESCRIPTION
**Pull request by dstandish**
Try to make clearer what this is doing by splitting things out a bit.

### CODE REVIEW

File: airflow/api_connexion/security.py [hunk 1]
```diff
@@ -113,18 +113,25 @@ def requires_access_dag(
...
+        def callback() -> bool | DagAccessEntity:
+            if dag_id:
+                # a DAG id is provided; is the user authorized to access this DAG?
+                return get_auth_manager().is_authorized_dag(
+                    method=method,
+                    access_entity=access_entity,
+                    details=DagDetails(id=dag_id),
+                )
+            else:
+                # here we know no DAG id is provided.
```
Comment (dstandish) [inline airflow/api_connexion/security.py]:
```suggestion
                # here we know dag_id is *not* provided.
```

File: airflow/api_connexion/security.py [hunk 2]
```diff
@@ -113,18 +113,25 @@ def requires_access_dag(
...
+        def callback() -> bool | DagAccessEntity:
+            if dag_id:
+                # a DAG id is provided; is the user authorized to access this DAG?
+                return get_auth_manager().is_authorized_dag(
+                    method=method,
+                    access_entity=access_entity,
+                    details=DagDetails(id=dag_id),
+                )
+            else:
+                # here we know dag_id is *not* provided.
```
Comment (dstandish) [inline airflow/api_connexion/security.py]:
```suggestion
                # here we know dag_id is not provided.
```

File: airflow/api_connexion/security.py [hunk 3]
```diff
@@ -113,18 +113,25 @@ def requires_access_dag(
...
+                    details=DagDetails(id=dag_id),
+                )
+            else:
+                # here we know dag_id is not provided.
+                # check is the user authorized to access all DAGs?
+                if access := get_auth_manager().is_authorized_dag(
+                    method=method,
+                    access_entity=access_entity,
+                ):
+                    return access
```
Comment (dstandish) [inline airflow/api_connexion/security.py]:
```suggestion
                    return True
```

File: airflow/api_connexion/security.py [hunk 4]
```diff
@@ -113,18 +113,25 @@ def requires_access_dag(
...
+                # a DAG id is provided; is the user authorized to access this DAG?
+                return get_auth_manager().is_authorized_dag(
+                    method=method,
+                    access_entity=access_entity,
+                    details=DagDetails(id=dag_id),
+                )
+            else:
+                # here we know dag_id is not provided.
+                # check is the user authorized to access all DAGs?
+                if access := get_auth_manager().is_authorized_dag(
```
Comment (dstandish) [inline airflow/api_connexion/security.py]:
```suggestion
                if get_auth_manager().is_authorized_dag(
```

**Comment by dstandish** [comment]
@vincbeck if you have a min to look tomorrow

**Review by vincbeck — APPROVED** [review]

**Pull request merged**